src/schema/v2/conversation/send_message_mutation.ts

```diff
@@ -60,35 +117,53 @@ export default mutationWithClientMutationId<any, any, ResolverContext>({
     },
   },
   mutateAndGetPayload: (
-    { id, from, bodyText: body_text, replyToMessageID: reply_to_message_id },
+    args,
     { conversationLoader, conversationCreateMessageLoader, userID }
   ) => {
-    if (!conversationCreateMessageLoader || !conversationLoader) return null
-    return conversationCreateMessageLoader(id, {
-      from,
-      reply_to_message_id,
-      body_text,
-      reply_all: true,
-      from_id: userID,
-    }).then(({ id: newMessageID }) => {
-      return conversationLoader(id).then((updatedConversation) => {
-        return {
-          conversation: updatedConversation,
-          // Because Impulse does not have the full new message object available immediately, we return an optimistic
-          // response so the mutation can return it too.
-          newMessagePayload: {
-            id: newMessageID,
-            from_email_address: from,
-            from_id: userID,
-            raw_text: body_text,
-            body: body_text,
-            created_at: new Date().toISOString(),
-            attachments: [],
-            // This addition is only for MP so it can determine if the message was from the current user.
-            conversation_from_address: updatedConversation.from_email,
-          },
-        }
-      })
+    if (!conversationCreateMessageLoader || !conversationLoader) {
+      return null
+    }
+
+    // The params replyAll and to are mutually exclusive. If replyAll: false
+    // and `to` is defined, set `replyAll: undefined` to avoid Impulse error.
+    const replyAll =
+      args.replyAll === false && args.to ? undefined : args.replyAll
+
+    return conversationCreateMessageLoader(args.id, {
+      attachments: args.attachments,
+      body_html: args.bodyHTML,
+      body_text: args.bodyText,
+      from: args.from,
+      from_id: args.fromId ?? userID,
+      reply_all: replyAll,
+      reply_to_message_id: args.replyToMessageID,
+      to: args.to,
     })
+      .then(({ id: newMessageID }) => {
+        return conversationLoader(args.id).then((updatedConversation) => {
+          return {
+            conversation: updatedConversation,
+
+            // Because Impulse does not have the full new message object available
+            // immediately, we return an optimistic response so the mutation can
+            // return it too.
+            newMessagePayload: {
+              attachments: args.attachments,
```

Comment:
`args.attachments` might be `null` or `undefined`, maybe needs a `|| []`?

Reply:
Oh good idea, will fix